lib/src/main/java/dev/openfeature/javasdk/OpenFeatureClient.java

```diff
@@ -84,6 +84,7 @@ <T> FlagEvaluationDetails<T> evaluateFlag(FlagValueType type, String key, T defa
             }
             details.value = defaultValue;
             details.reason = Reason.ERROR;
+            // We may want to extend exception and add a defined code field, instead of just using message.
```

Comment:
This is the only thing I think might need changing. Should we have a code field in a custom exception that's has it's own purpose beyond the general message?

I know this isn't in the spec, but maybe it should be...

cc @beeme1mr

Reply:
What would we hope to accomplish? The `code` is a useful tool for devs for debugging. We have strings which are useful to understand what went wrong. That's an exception message.

Can you share an example so I can better understand?

Reply:
I only mean that a dev might not assume the `message` gets put into the `errorCode` field. If we define a custom `OpenFeatureException`, with a `code` prop, we can say that the value of the code field will be put into the `errorCode` field in the `EvaluationDetails`.

Reply:
So if folks don't throw our special exception type, are you thinking that we just leave the error code empty?

Reply:
I think I'd like to leave this as is. Giving the original error provides something for the dev to work against when it comes to debugging. Just getting a "generic error" string back would be quite frustrating when something went wrong.